### PR TITLE
ActionBar component

### PR DESCRIPTION
**Pull request by mika-lindell**
- ActionBar & ActionBarItem -components
- ActionBarItem is only there for convenience, one should always use ActionBar to create necessary ActionBarItems

screen shot 2017-10-02 at 15 47 48

screen shot 2017-10-02 at 15 48 40

screen shot 2017-10-02 at 15 49 04

### CODE REVIEW

File: packages/hsl-shared-components/src/ActionBar/ActionBar.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import styled from 'styled-components';
+
+import ActionBarItem from './ActionBarItem';
+
+const ActionBar = styled(({ items, inverted, ...rest }) =>
+  (
+    <ul {...rest}>
+      {items.map((item, index) =>
+        (
+          <li style={{ display: 'flex', flex: '1' }}>
```
Comment (ktj) [inline packages/hsl-shared-components/src/ActionBar/ActionBar.js]: Styled components over inline styles

File: packages/hsl-shared-components/src/ActionBar/ActionBarItem.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import PropTypes from 'prop-types';
+import styled from 'styled-components/primitives';
+import { withTheme } from 'styled-components';
+import Touchable from '../Touchable';
+import View from '../View';
+import { P } from '../Typography';
+import { IS_NATIVE, size } from '../utils';
+
+const getTextColor = (active, inverted, theme) => {
+  if (inverted && !active) {
```
Comment (ktj) [inline packages/hsl-shared-components/src/ActionBar/ActionBarItem.js]:
```
if ((inverted && active) || (!inverted && !active)) { // if you want even shorter then if(inverted===active) {..
return theme.colors.primary.hslWhite;
}
return theme.font.colors.highlight;
```

File: packages/hsl-shared-components/src/ActionBar/ActionBarItem.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    return theme.font.colors.highlight;
+  }
+  if (active) {
+    return theme.colors.primary.hslWhite;
+  }
+  return theme.font.colors.highlight;
+};
+
+const getIconColor = (active, inverted, theme, override) => {
+  if (override && !active) {
```
Comment (ktj) [inline packages/hsl-shared-components/src/ActionBar/ActionBarItem.js]: override won't override if active is true?

File: packages/hsl-shared-components/src/ActionBar/ActionBarItem.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    return theme.colors.primary.hslWhite;
+  }
+  return theme.font.colors.highlight;
+};
+
+const getBorderColor = (active, inverted, theme) => {
+  if (inverted) {
+    return theme.colors.primary.hslWhite;
+  }
+  if (active) {
```
Comment (ktj) [inline packages/hsl-shared-components/src/ActionBar/ActionBarItem.js]: When colors are inverted then active does nothing?

Comment (mika-lindell) [reply packages/hsl-shared-components/src/ActionBar/ActionBarItem.js]: Yes

File: packages/hsl-shared-components/src/ActionBar/ActionBarItem.js (hunk 3 above)
Comment (mika-lindell) [reply packages/hsl-shared-components/src/ActionBar/ActionBarItem.js]: Yes that's correct, setting something as active needs to changes background color.
But you're right, maybe we should give the choice to progframmer – he'd just need to supply us with different icon for active state if its needed.

**Pull request merged**